File: python/Linked_Lists/19_RemoveNthNodeFromTheEnd..py
class ListNode:
    def __init__ (self,val=0,next=None):
        self.val = val
        self.next = next
class RemoveNthNodeFromTheEnd:
    def removeNthFromEnd(self, head, n):
        dummy = ListNode(0,head)
        result = head
        length=0
        if n == 0:
            return head
        while(result):
            length+=1
            result = result.next
        result = dummy
        length= length-n
        for i in range(length):
            result=result.next
        result.next=result.next.next
        return dummy.next

def createList():
    n = int(input("Enter the number of elements : "))
    head = ListNode()
    realhead = head
    print("Enter the elements : ")
    for i in range(n):
        k=int(input())
        a = ListNode(k,None)
        head.next=a
        head= head.next
    return realhead.next
def printList(head):
    while(head!=None):
        print(head.val,end=" ")
        head=head.next
def main():
    head = ListNode()
    head = createList()
    n = int(input("Enter the element to be deleted : "))
    obj = RemoveNthNodeFromTheEnd()
    head = obj.removeNthFromEnd(head,n)
    printList(head)

if __name__ == "__main__":
    main()

File: c/Linked_List/19_RemoveNthNodeFromTheEnd.c
#include<stdio.h>
#include<stdlib.h>

struct ListNode{
    int val;
    struct ListNode* next;
};
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    int length=0;
    struct ListNode* dummy = malloc(sizeof(struct ListNode));
    dummy->val = 0;
    dummy->next = head;
    struct ListNode* curr = head;
    while(curr){
        length++;
        curr=curr->next;
    }
    length=length-n;
    curr= dummy;
    for (int i =0 ; i < length ; i ++){
        curr= curr->next;
    }
    curr->next=curr->next->next;
    return dummy->next;
}

struct ListNode* createList(){
    printf("Enter the number of elements in the LinkedList : ");
    int n;
    scanf("%d",&n);
    struct ListNode* head = (struct ListNode*)malloc(sizeof(struct ListNode));
    head->next = NULL;
    printf("Enter the elements : ");
    struct ListNode* curr = head;
    for(int i = 0 ; i< n ; i++){
        struct ListNode* test= (struct ListNode*)malloc(sizeof(struct ListNode));
        scanf("%d",&test->val);
        test->next = NULL;
        curr->next = test;
        curr= curr->next;
    }
    return head->next;
}
void printList(struct ListNode* head){
    while(head){
        printf("%d -> ",head->val);
        head=head->next;
    }
}
int main(){
    struct ListNode* head = (struct ListNode*)malloc(sizeof(struct ListNode));
    head= createList();
    printf("Enter the element from the end to be deleted : ");
    int n;
    scanf("%d",&n);
    head = removeNthFromEnd(head,n);
    printList(head);
    return 0;

}